utils/uftrace-script.h

```diff
@@ -0,0 +1,54 @@
+#ifndef UFTRACE_UFTRACE_SCRIPT_H
+#define UFTRACE_UFTRACE_SCRIPT_H
+
+#ifndef __cplusplus
+#include <stdint.h>
+#include <stdbool.h>
+#include "utils/list.h"
+#else
+#include <cstdint>
+#include <cstdbool>
+#endif
+
+/* informantion passed during initialization */
+#define SCRIPT_INFO_MAJOR_VERSION 1
+#define SCRIPT_INFO_MINOR_VERSION 0
+struct script_info {
+	struct si_version {
+		uint8_t major;
+		uint8_t minor;
+	} ver;
```

Comment:
I thought about handling the major version change doesn't allow the previously compiled shared objects cannot be used.  But minor version change allows that the shared objects can be used in a higher version of uftrace.

In summary
- If major version is different, then uftrace rejects loading the shared object.
- If uftrace minor version is higher than the shared object, then it can use the shared object without a problem.
- If uftrace minor version is lower than the shared object, then uftrace rejects loading the shared object.

But I also think that we can simply use a single integer version because the layout will not be changed often and we don't actually distribute the plugin shared object binaries.

Reply:
I've just made it to use a single uftrace version for both struct.